### PR TITLE
add appId to useApplicationsApi

### PR DESCRIPTION
**Pull request by caravinci**
using the Application header as a test subject, @samrichca pointed out the hook was missing the capacity to pass data body and methods other than GET. This PR addresses that, as well as covering some technical debt I had introduced in the header (which account for the extra changes).

### CODE REVIEW

File: components/pages/Applications/ApplicationForm/index.tsx [hunk 1]
```diff
@@ -11,18 +11,11 @@ import DnaLoader from '@icgc-argo/uikit/DnaLoader';
...
-      url: `${API.APPLICATIONS}/${appId}`,
-    })
-      .then(({ data }: { data: Record<string, any> }) => setAppData(data))
-      .catch((error: Error) => {
-        // TODO dev logging, errors should not be shown to user
-        console.error(error);
-      });
-  }, []);
+    response && setAppData(response.data);
+  }, [response]);
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/index.tsx:18]: Preventing technical debt from using the wrong hook

File: global/hooks/useApplicationsAPI.tsx [hunk 2]
```diff
@@ -34,7 +35,7 @@ ApplicationsRequestData) => {
           sort: stringifySort(sort),
           states: stringifyStates(states),
         },
-        url: API.APPLICATIONS,
+        url: `${API.APPLICATIONS}/${appId}`,
```
Comment (caravinci) [inline global/hooks/useApplicationsAPI.tsx:38]: this handles both having and not the appId available

File: components/Loader.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { ReactElement } from 'react';
+import { css } from '@emotion/core';
+import DnaLoader from '@icgc-argo/uikit/DnaLoader';
+
+const Loader = (): ReactElement => (
```
Comment (caravinci) [inline components/Loader.tsx:5]: new self centring, very reusable component.

File: components/pages/Applications/ApplicationForm/index.tsx [hunk 4]
```diff
@@ -1,31 +1,22 @@
...
 
   return isLoading || Object.values(appData).length < 1 ? (
-    <DnaLoader />
+    <Loader />
```
Comment (caravinci) [inline components/pages/Applications/ApplicationForm/index.tsx:19]: better loader, because <3

**Review by samrichca — APPROVED** [review]

**Pull request merged**